src/Modules/Station/Components/PlaylistTabs/Playlist/Playlist.tsx

```diff
@@ -1,49 +1,66 @@
-import { Song } from 'Models/Song';
+import { PlaylistSong } from 'Models/Song';
 import * as React from 'react';
 import { Component } from 'react';
 import FlipMoveList from 'react-flip-move';
 import { Card, CardBody } from 'reactstrap';
 import { UserServices } from 'Services/Http/UserServices';
+import { SongServices } from 'Services/Http';
 import { IFavouriteItem } from '../Favourite/FavouriteItem';
 import './Playlist.scss';
 import { PlaylistItem } from './PlaylistItem';
 
-interface IPlayListStates {
-  playlist: any[];
-}
-
 interface IPlaylistProps {
-  playlist: any[];
+  playlist: PlaylistSong[];
+  stationId: String;
 }
 
 interface IFavoriteListProps {
   favoriteList: IFavouriteItem[];
 }
 
+interface IStates {
+  playlist: PlaylistSong[];
+  votingError: string;
+}
+
 type Iprops = IPlaylistProps & IFavoriteListProps;
 
-export class Playlist extends Component<Iprops, IPlayListStates> {
+export class Playlist extends Component<Iprops, IStates> {
   private userServices: UserServices;
-  constructor(props: any) {
+  private songServices: SongServices;
+
+  constructor(props: Iprops) {
     super(props);
+
     this.userServices = new UserServices();
+    this.songServices = new SongServices();
+
     this.state = {
       playlist: props.playlist,
+      votingError: '',
     };
-
-    this.upVote = this.upVote.bind(this);
-    this.downVote = this.downVote.bind(this);
-    this.addFavouriteSong = this.addFavouriteSong.bind(this);
   }
 
-  public upVote = () => {
-    alert('Up vote clicked!');
-    // TODO: Implemented upVote function
+  public upVote = (songId: String) => {
```

Comment:
It should be `songId: string` with `s` in lowercase.
Note that `String` more like Java syntax, you should use `string` with `s` in lower case in TypeScript syntax